In [1]:
import numpy as np
import pandas as pd
import warnings; warnings.filterwarnings( "ignore" ) # This code don't show any warnings only to make the notebook readable
import os

In [2]:
os.listdir()

['cleaned data',
 'Exotic Cars Analysis Assignment.pdf',
 'Exotic Cars Analysis-A.ipynb',
 'Exotic Cars Analysis_School.ipynb',
 'Exotic-Cars-.csv',
 'ml_Exotic_Cars_Analysis_4_Nr . 1.ipynb',
 'ml_Session_4_School_best.ipynb',
 'Sesstion_4_str.replace.ipynb',
 'visualtaion.ipynb']

In [3]:
data = pd.read_csv("Exotic-Cars-.csv")

In [4]:
data.sample(3)

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
758,Mercedes-Benz,AMG GT R,2021,4,577,516,3.5,"180,000"
858,BMW,M4,2022,3,503,479,3.8,"74,000"
250,Ferrari,Portofino,2022,3.9,591,413,3.3,"218,750"


# We fix first the car price

#### Man kann so machen aber wir verwenden andere Methode:

- cars_Price = []
- for price in data['Price (in USD)']:
    - price = price.replace(',','')
    - price = int(price)
    - cars_Price.append(price)

In [5]:
data["Price (in USD)"] = data["Price (in USD)"].str.replace("," , "")

In [6]:
data["Price (in USD)"] = data["Price (in USD)"].astype(int)

In [7]:
data["Price (in USD)"].dtype

dtype('int64')

In [8]:
data["Price (in USD)"]

0        101200
1        274390
2        333750
3        142700
4        298000
         ...   
1002    3000000
1003    2000000
1004    1000000
1005    2600000
1006    2400000
Name: Price (in USD), Length: 1007, dtype: int64

# Fix "Engine Size (L)" 

In [9]:
data["Engine Size (L)"].unique()

array(['3', '5.2', '3.9', '4', '4.4', '6.2', '3.8', '8', '5', '3.5',
       '4.7', '2', '2.9', '6', 'Electric', '6.5', '3.7', 'Electric Motor',
       '2.5', '1.5 + Electric', '6.8', '8.4', nan, '6.6', '7', '1.7',
       '3.3', '-', '6.7', '1.8', 'Electric (tri-motor)', '5.5',
       'Electric (93 kWh)', 'Electric (100 kWh)', 'Hybrid (4.0)', '4.6',
       '3.6', '1.5', 'Hybrid', '5.7', '2.0 (Electric)', '4.0 (Hybrid)',
       '0', '6.4', '6.3', '2.3'], dtype=object)

#### Fix here:
- nan
- Electric Motor
- '-'
- '0'
- Hybrid (4.0)
- Hybrid (4.0)

In [10]:
data["Engine Size (L)"].dtype

# dtype('O') Buchstabe O

dtype('O')

### We check for car "Engine Size (L)" if is NaN in Elecktric cars !

In [11]:
data[ data["Engine Size (L)"].isna() ]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
389,Tesla,Roadster,2022,NaN,10000+,0,1.9,200000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900
916,Tesla,Roadster,2022,NaN,"10,000+",NaN,1.9,200000


# First car, fix Tesla = Electric Models 

In [12]:
data[ data["Engine Size (L)"] == "-"]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
335,Tesla,Roadster,2022,-,1000+,-,1.9,200000


In [13]:
data[  data["Car Model"] == "Roadster"]

# hier ist bei ziffer ( 335 - ), ( 885 0 ) und (389 ,  916 NaN) nicht gut !

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000+,737,1.9,200000
335,Tesla,Roadster,2022,-,1000+,-,1.9,200000
354,Tesla,Roadster,2022,Electric,1000+,"10,000+",1.9,200000
364,Tesla,Roadster,2023,Electric,"1,000+",737,< 1.9,200000
389,Tesla,Roadster,2022,NaN,10000+,0,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,0,"10,000","7,376",1.9,200000
916,Tesla,Roadster,2022,NaN,"10,000+",NaN,1.9,200000


## Filtern wir nur diese - , NaN, 0

In [14]:
print(data.loc[335, "Engine Size (L)"])
print(data.loc[389, "Engine Size (L)"])
print(data.loc[885, "Engine Size (L)"])
print(data.loc[916, "Engine Size (L)"])

-
nan
0
nan


In [15]:
data.loc[ [335, 389, 885, 916],"Engine Size (L)"]

335      -
389    NaN
885      0
916    NaN
Name: Engine Size (L), dtype: object

In [16]:
data.loc[ [335, 389, 885, 916],"Engine Size (L)"] = "Electric"

# All numbers --> insert to "Electric"

In [17]:
data[  data["Car Model"] == "Roadster"]

# Chech all cars

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000+,737,1.9,200000
335,Tesla,Roadster,2022,Electric,1000+,-,1.9,200000
354,Tesla,Roadster,2022,Electric,1000+,"10,000+",1.9,200000
364,Tesla,Roadster,2023,Electric,"1,000+",737,< 1.9,200000
389,Tesla,Roadster,2022,Electric,10000+,0,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,Electric,"10,000","7,376",1.9,200000
916,Tesla,Roadster,2022,Electric,"10,000+",NaN,1.9,200000


In [18]:
data.loc[ [335, 389, 885, 916],"Engine Size (L)"]

335    Electric
389    Electric
885    Electric
916    Electric
Name: Engine Size (L), dtype: object

In [19]:
data["Engine Size (L)"].isna().sum()

# np.int64(8) = noch 8 Autos


np.int64(8)

In [20]:
data[ data["Engine Size (L)"].isna()]

# A Check of all NaN Electric cars

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900


In [21]:
data[ data["Engine Size (L)"].isna()]["Car Model"].unique()

# B Check of all NaN Electric cars

array(['C_Two', 'Model S Plaid', 'Taycan Turbo S', 'Evija', 'Taycan'],
      dtype=object)

In [22]:
data[  data["Car Model"] == "Model S Plaid" ]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190
320,Tesla,Model S Plaid,2021,Electric,1020,1050,1.9,131100
401,Tesla,Model S Plaid,2021,Electric (tri-motor),1020,1050,1.9,141190
511,Tesla,Model S Plaid,2021,Electric,1020,1050,1.9,141990
527,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,140000
540,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,139990
622,Tesla,Model S Plaid,2022,Electric (100 kWh),1020,1050,1.98,131190
642,Tesla,Model S Plaid,2021,Electric,1020,NaN,1.9,139990
965,Tesla,Model S Plaid,2022,Electric,"1,020","1,050",1.9,141190


# 2 Filter, both is the same !

In [23]:
data[ data["Engine Size (L)"].isna()][data["Car Model"] == "Model S Plaid" ]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190


In [24]:
data[ data["Engine Size (L)"].isna()][data["Car Model"] == "Model S Plaid" ].index

Index([171, 247], dtype='int64')

In [25]:
data[  data["Car Model"] == "Model S Plaid" ][data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190


In [26]:
model_S_Plaid_with_null_engine = data[ data["Car Model"] == "Model S Plaid" ][data["Engine Size (L)"].isna()].index
model_S_Plaid_with_null_engine

# index am Ende sehr wichtig für loc !
# Hier eine Variable erstellen und alle filter darauf speichern, welche NaN wo fehlen !

Index([171, 247], dtype='int64')

In [27]:
data.loc[ model_S_Plaid_with_null_engine]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,NaN,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,NaN,1020,1050,1.9,131190


In [28]:
data.loc[ model_S_Plaid_with_null_engine, "Engine Size (L)"]

171    NaN
247    NaN
Name: Engine Size (L), dtype: object

# --> Fix index to Electric

In [29]:
data.loc[ model_S_Plaid_with_null_engine, "Engine Size (L)"] = "Electric"

In [30]:
data.loc[ model_S_Plaid_with_null_engine, "Engine Size (L)"]

# Check

171    Electric
247    Electric
Name: Engine Size (L), dtype: object

#### Man hätte so machen können, aber die andere ist eine neue Methode !

- data.loc[[ 171, 247], "Engine Size (L)" ] = "Electric" <br>
--> To check !
- data.loc[[ 171, 247], "Engine Size (L)" ]


In [31]:
data["Engine Size (L)"].isna().sum()

# np.int64(6) jetzt nur noch 6 stk zum bereinigen !

np.int64(6)

In [32]:
data[ data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900


# --> Fix Rimac cars

In [33]:
data[ data["Car Model"] == "C_Two" ]

# Filter all C_Two car

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
278,Rimac,C_Two,2022,Electric,1914,1732,1.85,2400000
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
439,Rimac,C_Two,2021,Electric,1914,1732,1.8,2400000
509,Rimac,C_Two,2021,Electric,1914,1696,1.9,2400000
526,Rimac,C_Two,2022,Electric,1914,1696,1.9,2400000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000


In [34]:
data[ data["Engine Size (L)"].isna()][data["Car Model"] == "C_Two"]

# 2 Filter --> Filter only NaN Dataset in Rimac


,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000


In [35]:
C_Two_with_null_engine = data[data["Engine Size (L)"].isna()][data["Car Model"] == "C_Two"].index

# Am Ende index ist sehr wichtig für loc !!
# Eine Verable geben  und filter dazu speichern

In [36]:
C_Two_with_null_engine

# Check

Index([168, 387, 686], dtype='int64')

# --> Insert Rimac index to Electric

In [37]:
data.loc[ C_Two_with_null_engine ]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,NaN,1914,1696,1.9,2400000
387,Rimac,C_Two,2022,NaN,1888,1696,1.8,2400000
686,Rimac,C_Two,2022,NaN,1914,1696,1.85,2400000


In [38]:
data.loc[ C_Two_with_null_engine, "Engine Size (L)" ]

168    NaN
387    NaN
686    NaN
Name: Engine Size (L), dtype: object

In [39]:
data.loc[ C_Two_with_null_engine, "Engine Size (L)" ] = "Electric"

In [40]:
data.loc[ C_Two_with_null_engine ]

# Check, ok

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
168,Rimac,C_Two,2022,Electric,1914,1696,1.9,2400000
387,Rimac,C_Two,2022,Electric,1888,1696,1.8,2400000
686,Rimac,C_Two,2022,Electric,1914,1696,1.85,2400000


# Check other car for "Engine Size (L)" = NaN

In [41]:
data["Engine Size (L)"].isna().sum()

# Still 3 cars to fix !

np.int64(3)

In [42]:
data["Engine Size (L)"].nunique()

43

# All car still habe NaN

In [43]:
data[ data["Engine Size (L)"].isna()]

# Filter all car for NaN

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900


# Fix and filter all Taycan Turbo S

In [44]:
data[data["Car Model"]== "Taycan Turbo S"]

# Filter all  Taycan Turbo S

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000
305,Porsche,Taycan Turbo S,2021,Electric,750,774,2.6,185000
543,Porsche,Taycan Turbo S,2022,Electric,750,774,2.6,185000
717,Porsche,Taycan Turbo S,2022,Electric,750,774,2.6,185000
869,Porsche,Taycan Turbo S,2021,Electric,750,774,2.6,185000
955,Porsche,Taycan Turbo S,2021,Electric,750,774,2.6,185000


In [45]:
data[data["Car Model"]== "Taycan Turbo S"][data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000


In [46]:
data[data["Car Model"]== "Taycan Turbo S"][data["Engine Size (L)"].isna()].index

Index([222], dtype='int64')

In [47]:
Taycan_Turbo_S_with_null_engine = data[data["Car Model"] == "Taycan Turbo S"][data["Engine Size (L)"].isna()].index

Taycan_Turbo_S_with_null_engine

Index([222], dtype='int64')

In [48]:
data.loc[Taycan_Turbo_S_with_null_engine, ]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
222,Porsche,Taycan Turbo S,2021,NaN,750,774,2.6,185000


In [49]:
data.loc[Taycan_Turbo_S_with_null_engine, "Engine Size (L)" ]

222    NaN
Name: Engine Size (L), dtype: object

In [50]:
data.loc[Taycan_Turbo_S_with_null_engine, "Engine Size (L)" ] = "Electric"

In [51]:
data.loc[Taycan_Turbo_S_with_null_engine, ]

# Check 

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
222,Porsche,Taycan Turbo S,2021,Electric,750,774,2.6,185000


# Fix other Porsche for NaN

In [52]:
data[ data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900


In [53]:
data[ data["Car Model"] == "Taycan"][data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
752,Porsche,Taycan,2022,NaN,469,479,3.8,79900


In [54]:
data[ data["Car Model"] == "Taycan"][data["Engine Size (L)"].isna()].index

Index([752], dtype='int64')

In [55]:
Taycan_with_null_engine = data[ data["Car Model"] == "Taycan"][data["Engine Size (L)"].isna()].index

Taycan_with_null_engine

Index([752], dtype='int64')

In [56]:
data.loc [ Taycan_with_null_engine, "Engine Size (L)" ] = "Electric"

In [57]:
data.loc [ Taycan_with_null_engine, "Engine Size (L)" ]

# Check

752    Electric
Name: Engine Size (L), dtype: object

# Check again

In [58]:
data["Engine Size (L)"].isna().sum()

np.int64(1)

In [59]:
data[ data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000


# Fix Lotus Evija

In [60]:
data[ data["Car Model"] == "Evija"]

# We have here tow (2) 697 and 1003 Model

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
420,Lotus,Evija,2022,Electric,1973,1254,2.5,2750000
523,Lotus,Evija,2022,Electric,1973,1254,2.5,2600000
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000
877,Lotus,Evija,2021,Electric,2000,1254,2.8,2800000
987,Lotus,Evija,2022,Electric,1973,1254,2.5,2000000
1003,Lotus,Evija,2021,Electric Motor,1972,1254,2,2000000


In [61]:
data[ data["Car Model"] == "Evija"][data["Engine Size (L)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
697,Lotus,Evija,2022,NaN,1972,1254,2.5,2700000


In [62]:
Lotus_with_null_engine = data[ data["Car Model"] == "Evija"][data["Engine Size (L)"].isna()].index

In [63]:
data.loc[Lotus_with_null_engine, "Engine Size (L)" ] = "Electric"

In [64]:
data.loc[Lotus_with_null_engine, "Engine Size (L)" ]

# Check ! ok

697    Electric
Name: Engine Size (L), dtype: object

In [65]:
data[ data["Car Model"] == "Evija"]

# Last index 1003, Fix it 

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
420,Lotus,Evija,2022,Electric,1973,1254,2.5,2750000
523,Lotus,Evija,2022,Electric,1973,1254,2.5,2600000
697,Lotus,Evija,2022,Electric,1972,1254,2.5,2700000
877,Lotus,Evija,2021,Electric,2000,1254,2.8,2800000
987,Lotus,Evija,2022,Electric,1973,1254,2.5,2000000
1003,Lotus,Evija,2021,Electric Motor,1972,1254,2,2000000


In [66]:
data.loc[ 1003, "Engine Size (L)"] = "Electric"

In [67]:
data[ data["Car Model"] == "Evija"]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
420,Lotus,Evija,2022,Electric,1973,1254,2.5,2750000
523,Lotus,Evija,2022,Electric,1973,1254,2.5,2600000
697,Lotus,Evija,2022,Electric,1972,1254,2.5,2700000
877,Lotus,Evija,2021,Electric,2000,1254,2.8,2800000
987,Lotus,Evija,2022,Electric,1973,1254,2.5,2000000
1003,Lotus,Evija,2021,Electric,1972,1254,2,2000000


In [68]:
data["Engine Size (L)"].isna().sum()

np.int64(0)

In [69]:
data["Engine Size (L)"].nunique()

43

In [70]:
data["Engine Size (L)"].unique()

array(['3', '5.2', '3.9', '4', '4.4', '6.2', '3.8', '8', '5', '3.5',
       '4.7', '2', '2.9', '6', 'Electric', '6.5', '3.7', 'Electric Motor',
       '2.5', '1.5 + Electric', '6.8', '8.4', '6.6', '7', '1.7', '3.3',
       '6.7', '1.8', 'Electric (tri-motor)', '5.5', 'Electric (93 kWh)',
       'Electric (100 kWh)', 'Hybrid (4.0)', '4.6', '3.6', '1.5',
       'Hybrid', '5.7', '2.0 (Electric)', '4.0 (Hybrid)', '6.4', '6.3',
       '2.3'], dtype=object)

In [71]:
data[ data["Engine Size (L)"] == "Electric Motor"]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
37,Porsche,Taycan 4S,2022,Electric Motor,562,479,3.8,104000
1006,Rimac,Nevera,2021,Electric Motor,1888,1696,1.85,2400000


In [72]:
data.loc[ [ 37, 1006], "Engine Size (L)" ] = "Electric"

In [73]:
data["Engine Size (L)"][[37, 1006]]

37      Electric
1006    Electric
Name: Engine Size (L), dtype: object

In [74]:
data["Engine Size (L)"].unique()

array(['3', '5.2', '3.9', '4', '4.4', '6.2', '3.8', '8', '5', '3.5',
       '4.7', '2', '2.9', '6', 'Electric', '6.5', '3.7', '2.5',
       '1.5 + Electric', '6.8', '8.4', '6.6', '7', '1.7', '3.3', '6.7',
       '1.8', 'Electric (tri-motor)', '5.5', 'Electric (93 kWh)',
       'Electric (100 kWh)', 'Hybrid (4.0)', '4.6', '3.6', '1.5',
       'Hybrid', '5.7', '2.0 (Electric)', '4.0 (Hybrid)', '6.4', '6.3',
       '2.3'], dtype=object)

In [75]:
data["Engine Size (L)"].nunique()

# 42 is ok

42

# Fix Horsepower with str = STR

In [76]:
data["Horsepower"].isna().sum()

np.int64(0)

In [77]:
data["Horsepower"].unique()

# Fix 10000+, 10,000, 1000+, 1,500, 1,020

array(['379', '630', '661', '562', '710', '617', '523', '490', '760',
       '600', '1500', '717', '296', '1280', '471', '416', '454', '300',
       '505', '320', '626', '671', '622', '720', '1914', '414', '759',
       '986', '591', '503', '650', '660', '350', '641', '611', '394',
       '612', '369', '603', '455', '460', '325', '349', '592', '444',
       '405', '797', '770', '332', '473', '480', '573', '380', '1600',
       '181', '620', '764', '624', '1000+', '382', '800', '715', '690',
       '730', '469', '365', '401', '645', '435', '1020', '500', '780',
       '750', '402', '575', '729', '789', '577', '495', '237', '310',
       '791', '1874', '542', '368', '616', '1479', '755', '1,000+', '288',
       '1888', '10000+', '482', '1973', '1262', '1035', '819', '385',
       '647', '1200', '1578', '625', '583', '429', '563', '400', '707',
       '887', '1972', '305', '640', '255', '689', '372', '1000', '2000',
       '550', '10,000', '1,500', '10,000+', '485', '1,020', '1872', '621'

In [78]:
data[ data["Horsepower"] == "10000+"]

# First filter 10000+ 

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
389,Tesla,Roadster,2022,Electric,10000+,0,1.9,200000


In [79]:
data[ data["Car Model"] == "Roadster"]

# Filter all Roadster

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000+,737,1.9,200000
335,Tesla,Roadster,2022,Electric,1000+,-,1.9,200000
354,Tesla,Roadster,2022,Electric,1000+,"10,000+",1.9,200000
364,Tesla,Roadster,2023,Electric,"1,000+",737,< 1.9,200000
389,Tesla,Roadster,2022,Electric,10000+,0,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,Electric,"10,000","7,376",1.9,200000
916,Tesla,Roadster,2022,Electric,"10,000+",NaN,1.9,200000


# Direct => better !
- data['Horsepower']=data['Horsepower'].str.replace( ',' , '')
- All ',' to nothing ''
- data['Horsepower']=data['Horsepower'].str.replace( '+' , '')
- All '+' to nothing ''

In [80]:
data[ "Horsepower"] = data[ "Horsepower"].str.replace("1000+", "1000")
data[ "Horsepower"] = data[ "Horsepower"].str.replace("1,000", "1000")
data[ "Horsepower"] = data[ "Horsepower"].str.replace("10000+", "1000")
data[ "Horsepower"] = data[ "Horsepower"].str.replace("10000+", "1000")
data[ "Horsepower"] = data[ "Horsepower"].str.replace("10,000", "1000")
data[ "Horsepower"] = data[ "Horsepower"].str.replace("10,000+", "1000")

# Change all together

In [81]:
data[ data["Car Model"] == "Roadster"]

# Check

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
335,Tesla,Roadster,2022,Electric,1000,-,1.9,200000
354,Tesla,Roadster,2022,Electric,1000,"10,000+",1.9,200000
364,Tesla,Roadster,2023,Electric,1000+,737,< 1.9,200000
389,Tesla,Roadster,2022,Electric,1000,0,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,Electric,1000,"7,376",1.9,200000
916,Tesla,Roadster,2022,Electric,1000+,NaN,1.9,200000


In [82]:
data[ data["Horsepower"] == "1,500"]

# Filter all 1,500

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
898,Bugatti,Chiron,2021,8,"1,500","1,180",2.4,3000000


In [83]:
data["Horsepower"] = data["Horsepower"].str.replace("1,500", "1500")

# Change

In [84]:
data["Horsepower"] [898]

# Check

'1500'

In [85]:
data[ data["Horsepower"] == "1,020"]

# Filter 1,020

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
965,Tesla,Model S Plaid,2022,Electric,"1,020","1,050",1.9,141190


In [86]:
data["Horsepower"] = data["Horsepower"].str.replace("1,020", "1020")

# Change by str

In [87]:
data["Horsepower"][965]

# check by index

'1020'

In [88]:
data["Horsepower"].unique()

# Check, now is clean !

array(['379', '630', '661', '562', '710', '617', '523', '490', '760',
       '600', '1500', '717', '296', '1280', '471', '416', '454', '300',
       '505', '320', '626', '671', '622', '720', '1914', '414', '759',
       '986', '591', '503', '650', '660', '350', '641', '611', '394',
       '612', '369', '603', '455', '460', '325', '349', '592', '444',
       '405', '797', '770', '332', '473', '480', '573', '380', '1600',
       '181', '620', '764', '624', '1000', '382', '800', '715', '690',
       '730', '469', '365', '401', '645', '435', '1020', '500', '780',
       '750', '402', '575', '729', '789', '577', '495', '237', '310',
       '791', '1874', '542', '368', '616', '1479', '755', '1000+', '288',
       '1888', '482', '1973', '1262', '1035', '819', '385', '647', '1200',
       '1578', '625', '583', '429', '563', '400', '707', '887', '1972',
       '305', '640', '255', '689', '372', '2000', '550', '485', '1872',
       '621'], dtype=object)

# Fix all "0-60 MPH Time (seconds)"

In [89]:
data["0-60 MPH Time (seconds)"].unique()

# Fix < 1.9

array(['4', '2.8', '3', '3.2', '2.7', '3.1', '3.8', '3.5', '2.5', '2.4',
       '5.4', '4.4', '4.8', '4.7', '3.6', '4.1', '1.85', '4.5', '3.3',
       '3.9', '4.2', '3.4', '5.1', '4.3', '2.9', '5', '5.3', '4.9', '6.5',
       '3.7', '1.95', '1.9', '1.98', '2.6', '2.3', '4.6', '< 1.9', '1.8',
       '2.1', '5.2', '2.2', '6.4', '2'], dtype=object)

In [90]:
data[ data["0-60 MPH Time (seconds)"] == "< 1.9" ]

# Filter < 1.9

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
364,Tesla,Roadster,2023,Electric,1000+,737,< 1.9,200000


In [91]:
data["0-60 MPH Time (seconds)"] = data["0-60 MPH Time (seconds)"].str.replace("< 1.9", "1.9" )

# Change


In [92]:
data["0-60 MPH Time (seconds)"][364]

# Check

'1.9'

In [93]:
data["0-60 MPH Time (seconds)"].unique()

array(['4', '2.8', '3', '3.2', '2.7', '3.1', '3.8', '3.5', '2.5', '2.4',
       '5.4', '4.4', '4.8', '4.7', '3.6', '4.1', '1.85', '4.5', '3.3',
       '3.9', '4.2', '3.4', '5.1', '4.3', '2.9', '5', '5.3', '4.9', '6.5',
       '3.7', '1.95', '1.9', '1.98', '2.6', '2.3', '4.6', '1.8', '2.1',
       '5.2', '2.2', '6.4', '2'], dtype=object)

In [94]:
data["0-60 MPH Time (seconds)"] = data["0-60 MPH Time (seconds)"].astype(float)

In [95]:
data.dtypes

# Horsepower = object

Car Make                    object
Car Model                   object
Year                         int64
Engine Size (L)             object
Horsepower                  object
Torque (lb-ft)              object
0-60 MPH Time (seconds)    float64
Price (in USD)               int64
dtype: object

In [96]:
data["0-60 MPH Time (seconds)"].unique()

# Check agin

array([4.  , 2.8 , 3.  , 3.2 , 2.7 , 3.1 , 3.8 , 3.5 , 2.5 , 2.4 , 5.4 ,
       4.4 , 4.8 , 4.7 , 3.6 , 4.1 , 1.85, 4.5 , 3.3 , 3.9 , 4.2 , 3.4 ,
       5.1 , 4.3 , 2.9 , 5.  , 5.3 , 4.9 , 6.5 , 3.7 , 1.95, 1.9 , 1.98,
       2.6 , 2.3 , 4.6 , 1.8 , 2.1 , 5.2 , 2.2 , 6.4 , 2.  ])

# Fix in Dataset: Torque (lb-ft)

In [97]:
data["Torque (lb-ft)"].isna().sum()

np.int64(3)

In [98]:
data[ data["Torque (lb-ft)"].isna()]

# Filter all NaN

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
642,Tesla,Model S Plaid,2021,Electric,1020,NaN,1.9,139990
878,Maserati,GranTurismo,2021,Electric,550,NaN,2.8,200000
916,Tesla,Roadster,2022,Electric,1000+,NaN,1.9,200000


In [99]:
data[ data["Car Model"] == "Model S Plaid"]

# Filter all Model S Plaid

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,Electric,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,Electric,1020,1050,1.90,131190
320,Tesla,Model S Plaid,2021,Electric,1020,1050,1.90,131100
401,Tesla,Model S Plaid,2021,Electric (tri-motor),1020,1050,1.90,141190
511,Tesla,Model S Plaid,2021,Electric,1020,1050,1.90,141990
527,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,140000
540,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,139990
622,Tesla,Model S Plaid,2022,Electric (100 kWh),1020,1050,1.98,131190
642,Tesla,Model S Plaid,2021,Electric,1020,NaN,1.90,139990
965,Tesla,Model S Plaid,2022,Electric,1020,"1,050",1.90,141190


In [100]:
data["Torque (lb-ft)"] = data["Torque (lb-ft)"].str.replace("1,050", "1050")
data.loc[ 642,"Torque (lb-ft)" ] = "1050"

In [101]:
data[ data["Car Model"] == "Model S Plaid"]

# check for Model S Plaid

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
171,Tesla,Model S Plaid,2021,Electric,1020,1050,1.98,131190
247,Tesla,Model S Plaid,2022,Electric,1020,1050,1.90,131190
320,Tesla,Model S Plaid,2021,Electric,1020,1050,1.90,131100
401,Tesla,Model S Plaid,2021,Electric (tri-motor),1020,1050,1.90,141190
511,Tesla,Model S Plaid,2021,Electric,1020,1050,1.90,141990
527,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,140000
540,Tesla,Model S Plaid,2022,Electric,1020,1050,1.98,139990
622,Tesla,Model S Plaid,2022,Electric (100 kWh),1020,1050,1.98,131190
642,Tesla,Model S Plaid,2021,Electric,1020,1050,1.90,139990
965,Tesla,Model S Plaid,2022,Electric,1020,1050,1.90,141190


In [102]:
data[ data["Torque (lb-ft)"].isna()]

# check for other open NaN



,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
878,Maserati,GranTurismo,2021,Electric,550,NaN,2.8,200000
916,Tesla,Roadster,2022,Electric,1000+,NaN,1.9,200000


In [103]:
data[ data["Car Model"]== "GranTurismo"][data["Year"] == 2021]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
17,Maserati,GranTurismo,2021,4.7,454,384,4.8,150000
70,Maserati,GranTurismo,2021,4.7,454,384,4.7,150980
288,Maserati,GranTurismo,2021,4.7,454,384,4.8,134300
347,Maserati,GranTurismo,2021,4.7,454,384,4.7,133050
362,Maserati,GranTurismo,2021,4.7,454,384,4.8,132400
382,Maserati,GranTurismo,2021,4.7,454,384,4.8,150980
480,Maserati,GranTurismo,2021,4.7,454,384,4.7,134300
524,Maserati,GranTurismo,2021,4.7,454,384,4.8,134300
602,Maserati,GranTurismo,2021,4.7,454,384,4.7,134300
621,Maserati,GranTurismo,2021,4.7,454,384,4.7,150980


In [104]:
data.loc[ 878, "Torque (lb-ft)" ] = 384

In [105]:
data[ data["Car Model"]== "GranTurismo"][data["Year"] == 2021]

# check all car Model, GranTurismo > Year = 2021

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
17,Maserati,GranTurismo,2021,4.7,454,384,4.8,150000
70,Maserati,GranTurismo,2021,4.7,454,384,4.7,150980
288,Maserati,GranTurismo,2021,4.7,454,384,4.8,134300
347,Maserati,GranTurismo,2021,4.7,454,384,4.7,133050
362,Maserati,GranTurismo,2021,4.7,454,384,4.8,132400
382,Maserati,GranTurismo,2021,4.7,454,384,4.8,150980
480,Maserati,GranTurismo,2021,4.7,454,384,4.7,134300
524,Maserati,GranTurismo,2021,4.7,454,384,4.8,134300
602,Maserati,GranTurismo,2021,4.7,454,384,4.7,134300
621,Maserati,GranTurismo,2021,4.7,454,384,4.7,150980


In [106]:
data["Torque (lb-ft)"][878]

# short check !

384

In [107]:
data[ data["Torque (lb-ft)"].isna()]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
916,Tesla,Roadster,2022,Electric,1000+,NaN,1.9,200000


In [108]:
data [ data["Car Model"] == "Roadster"]

# Fix all "-" , 10,000+ , 0, 7,376 und NaN

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
335,Tesla,Roadster,2022,Electric,1000,-,1.9,200000
354,Tesla,Roadster,2022,Electric,1000,"10,000+",1.9,200000
364,Tesla,Roadster,2023,Electric,1000+,737,1.9,200000
389,Tesla,Roadster,2022,Electric,1000,0,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,Electric,1000,"7,376",1.9,200000
916,Tesla,Roadster,2022,Electric,1000+,NaN,1.9,200000


In [109]:
data["Torque (lb-ft)"] = data["Torque (lb-ft)"].str.replace("-", "737")
data["Torque (lb-ft)"] = data["Torque (lb-ft)"].str.replace("10,000+", "737")
data["Torque (lb-ft)"] = data["Torque (lb-ft)"].str.replace("0", "737")
data["Torque (lb-ft)"] = data["Torque (lb-ft)"].str.replace("7,376", "737")
data.loc[ 916, "Torque (lb-ft)"] = 737

In [110]:
data [ data["Car Model"] == "Roadster"]

,Car Make,Car Model,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
99,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
335,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
354,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
364,Tesla,Roadster,2023,Electric,1000+,737,1.9,200000
389,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
821,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
885,Tesla,Roadster,2022,Electric,1000,737,1.9,200000
916,Tesla,Roadster,2022,Electric,1000+,737,1.9,200000


In [111]:
data["Torque (lb-ft)"].isna().sum()

np.int64(1)

# Distribute all cars according to origin country

In [112]:
data["Car Make"].unique()

array(['Porsche', 'Lamborghini', 'Ferrari', 'Audi', 'McLaren', 'BMW',
       'Mercedes-Benz', 'Chevrolet', 'Ford', 'Nissan', 'Aston Martin',
       'Bugatti', 'Dodge', 'Jaguar', 'Koenigsegg', 'Lexus', 'Lotus',
       'Maserati', 'Alfa Romeo', 'Ariel', 'Bentley', 'Mercedes-AMG',
       'Pagani', 'Polestar', 'Rimac', 'Acura', 'Mazda', 'Rolls-Royce',
       'Tesla', 'Toyota', 'W Motors', 'Shelby', 'TVR', 'Subaru',
       'Pininfarina', 'Kia', 'Alpine', 'Ultima'], dtype=object)

In [113]:
cars_country_list = []

for car in data['Car Make'] :

    if car in [ 'Porsche', 'Audi', 'BMW', 'Mercedes-Benz', 'Mercedes-AMG',  ] :
        cars_country_list.append( 'German' )

    elif car in [ 'Lamborghini', 'Ferrari', 'Maserati', 'Alfa Romeo', 'Pagani', 'Pininfarina'] :
         cars_country_list.append( 'Italian' ) 

    elif car in ["McLaren", "Aston Martin", "Jaguar", "Lotus", "Bentley","Rolls-Royce", "Ariel", "TVR", "Ultima"]:  
        cars_country_list.append( 'UK' )  
    
    elif car in ["Chevrolet", "Ford", "Dodge", "Tesla", "Shelby"]:  
        cars_country_list.append( 'US' )   

    elif car in ["Nissan", "Lexus", "Acura", "Mazda", "Toyota", "Subaru"]:  
        cars_country_list.append( 'Japan' )  

    elif car in ["Bugatti", "Alpine"]:  
        cars_country_list.append( 'France' )     

    elif car in ["Koenigsegg", "Polestar"]:  
        cars_country_list.append( 'Sweden' )  

    elif car in [ "Rimac"]:  
        cars_country_list.append( 'Croatia' )  

    elif car in ["Kia"]:  
        cars_country_list.append( 'South Korea' )  

    elif car in ["W Motors"]:  
        cars_country_list.append( 'United Arab Emirates' )  

    else :
       cars_country_list.append( 'Other' )        
        

In [114]:
data.insert(2, "Country", cars_country_list)

In [115]:
data.sample(2)

,Car Make,Car Model,Country,Year,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
936,Aston Martin,Vantage,UK,2022,4,503,57375,3.5,149995
760,Ford,GT,US,2022,3.5,660,55737,3.0,500000


In [116]:
data.sample(10)[["Car Make", "Country"]]

,Car Make,Country
833,Mercedes-Benz,German
483,BMW,German
227,BMW,German
203,Nissan,Japan
172,Toyota,Japan
180,BMW,German
881,Porsche,German
543,Porsche,German
490,Nissan,Japan
957,Lamborghini,Italian


In [117]:
data["Country"].value_counts()

Country
German                  287
UK                      229
Italian                 176
US                      169
Japan                    88
France                   24
Sweden                   16
Croatia                  14
United Arab Emirates      3
South Korea               1
Name: count, dtype: int64

# --> PDF Exotic Cars Analysis Assignment


In [118]:
New_Segment = []

for price in data['Price (in USD)']:
    if price < 70000:
        New_Segment.append('Lower')
    
    elif price < 120000:
        New_Segment.append('Lower Mainstream')

    elif price < 180000:
        New_Segment.append('Mainstream')

    elif price < 250000:
        New_Segment.append('Upper Mainstream')

    elif price < 400000:
        New_Segment.append('Premium')

    elif price < 800000:
        New_Segment.append('Luxury')
    else:
        New_Segment.append('Exotic')


In [119]:
data.insert(4, 'New_Segment', New_Segment)

In [120]:
data.sample(10)[[  'Car Make', 'Country', 'New_Segment', 'Price (in USD)']]

,Car Make,Country,New_Segment,Price (in USD)
493,Ferrari,Italian,Premium,274280
422,Mercedes-Benz,German,Premium,326050
12,Dodge,US,Lower,61000
104,Ferrari,Italian,Luxury,625000
996,Mercedes-Benz,German,Premium,254500
501,Jaguar,UK,Lower Mainstream,105900
577,Alfa Romeo,Italian,Lower Mainstream,74750
419,Lexus,Japan,Lower Mainstream,93050
871,BMW,German,Lower Mainstream,71800
160,Jaguar,UK,Lower,69900


In [121]:
data['New_Segment'].value_counts()

New_Segment
Lower Mainstream    248
Lower               230
Upper Mainstream    180
Mainstream           96
Luxury               90
Exotic               83
Premium              80
Name: count, dtype: int64

In [122]:
data['New_Segment'].value_counts()

New_Segment
Lower Mainstream    248
Lower               230
Upper Mainstream    180
Mainstream           96
Luxury               90
Exotic               83
Premium              80
Name: count, dtype: int64

In [123]:
data[ data['New_Segment']== 'Exotic']

,Car Make,Car Model,Country,Year,New_Segment,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
11,Bugatti,Chiron,France,2021,Exotic,8,1500,118737,2.40,3000000
14,Koenigsegg,Jesko,Sweden,2021,Exotic,5,1280,173715,2.50,2800000
24,Pagani,Huayra,Italian,2021,Exotic,6,720,737,2.80,2800000
26,Rimac,Nevera,Croatia,2022,Exotic,Electric,1914,1696,1.85,2400000
85,Bugatti,Chiron,France,2022,Exotic,8,1500,118737,2.40,3000000
...,...,...,...,...,...,...,...,...,...,...
1002,Koenigsegg,Jesko,Sweden,2022,Exotic,5,1280,117376,2.50,3000000
1003,Lotus,Evija,UK,2021,Exotic,Electric,1972,1254,2.00,2000000
1004,McLaren,Senna,UK,2021,Exotic,4,789,59737,2.70,1000000
1005,Pagani,Huayra,Italian,2021,Exotic,6,764,738,3.00,2600000


In [124]:
data[ data['New_Segment']== 'Premium']

,Car Make,Car Model,Country,Year,New_Segment,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
1,Lamborghini,Huracan,Italian,2021,Premium,5.2,630,443,2.8,274390
2,Ferrari,488 GTB,Italian,2022,Premium,3.9,661,561,3.0,333750
4,McLaren,720S,UK,2021,Premium,4,710,568,2.7,298000
78,Ferrari,F8 Tributo,Italian,2021,Premium,3.9,710,568,2.8,274280
98,Rolls-Royce,Wraith,UK,2021,Premium,6.8,624,67375,4.4,330000
...,...,...,...,...,...,...,...,...,...,...
972,Lamborghini,Huracan,Italian,2021,Premium,5.2,630,443,2.8,274390
973,Ferrari,488 GTB,Italian,2022,Premium,3.9,661,561,3.0,333750
975,McLaren,720S,UK,2021,Premium,4,710,568,2.7,298000
991,Lamborghini,Aventador,Italian,2021,Premium,6.5,730,57379,2.9,393695


In [125]:
data[ data['New_Segment']== 'Luxury']

,Car Make,Car Model,Country,Year,New_Segment,Engine Size (L),Horsepower,Torque (lb-ft),0-60 MPH Time (seconds),Price (in USD)
28,Lamborghini,Aventador SVJ,Italian,2021,Luxury,6.5,759,531,2.8,517770
29,Ferrari,SF90 Stradale,Italian,2021,Luxury,4,986,59737,2.5,625000
35,Ford,GT,US,2022,Luxury,3.5,660,55737,3.0,500000
45,Lamborghini,Aventador SVJ,Italian,2021,Luxury,6.5,759,531,2.8,518000
46,Ferrari,SF90 Stradale,Italian,2021,Luxury,4,986,59737,2.5,625000
...,...,...,...,...,...,...,...,...,...,...
925,Ford,GT,US,2022,Luxury,3.5,660,55737,2.8,500000
957,Lamborghini,Aventador SVJ,Italian,2021,Luxury,6.5,759,531,2.8,517770
958,Ford,GT,US,2021,Luxury,3.5,660,55737,3.0,500000
966,Ferrari,SF90 Stradale,Italian,2021,Luxury,4,986,59737,2.5,625000


In [126]:
os.listdir()

['cleaned data',
 'Exotic Cars Analysis Assignment.pdf',
 'Exotic Cars Analysis-A.ipynb',
 'Exotic Cars Analysis_School.ipynb',
 'Exotic-Cars-.csv',
 'ml_Exotic_Cars_Analysis_4_Nr . 1.ipynb',
 'ml_Session_4_School_best.ipynb',
 'Sesstion_4_str.replace.ipynb',
 'visualtaion.ipynb']

In [129]:
data.to_csv('cleaned data/Cleaned_Exotic-Cars-.csv', index= False)